src/main/kotlin/com/hedvig/accountservice/termination/TerminationService.kt

```diff
@@ -69,21 +91,107 @@ class TerminationService @Autowired constructor(
         }
     }
 
-    private fun terminateProductOnLastDayOfMonth(
+    private fun notifyAndMaybeTerminateMember(
+        memberId: String,
+        requestId: String,
+        timestamp: Instant,
+        notification: (memberId: String, requestId: String) -> Unit
+    ) {
+        val contracts = productPricingService.getContractsByMemberId(memberId)
```

Comment:
Maybe filter out only active contracts for clarity?
```suggestion
        val contracts = productPricingService.getContractsByMemberId(memberId).filter(contract -> isActive(contract))
```